apps/src/redux.js

```diff
@@ -1,21 +1,124 @@
+/* eslint no-unused-vars: "error" */
 /** @file Code.org configured store-creation method.
  *  @see http://redux.js.org/docs/api/createStore.html */
 import Immutable from 'immutable';
 import experiments from "./util/experiments";
+import * as redux from 'redux';
+import reduxThunk from 'redux-thunk';
+import { connect } from 'react-redux';
 
-var redux = require('redux');
-var reduxThunk = require('redux-thunk').default;
 if (process.env.NODE_ENV !== "production") {
   var createLogger = require('redux-logger');
 }
 
+
+let reduxStore;
+let globalReducers = {};
+
+let __oldReduxStore;
+let __oldGlobalReducers;
+
+export function stubRedux() {
+  if (__oldReduxStore) {
+    throw new Error("Reduce store has already been stubbed. Did you forget to call restore?");
+  }
+  __oldReduxStore = reduxStore;
+  __oldGlobalReducers = globalReducers;
+  reduxStore = null;
+  globalReducers = {};
+}
+
+export function restoreRedux() {
+  reduxStore = __oldReduxStore;
+  globalReducers = __oldGlobalReducers;
+  __oldReduxStore = null;
+  __oldGlobalReducers = null;
+}
+
+
+/**
+ * Get a reference to our redux store. If it doesn't exist yet, create it.
+ */
+export function getStore() {
+  if (!reduxStore) {
+    reduxStore = createStore(redux.combineReducers(globalReducers));
+  }
+
+  return reduxStore;
+}
+
+/**
+ * Register a top-level reducer with the global store and get back
+ * a selector function to access the state for that reducer.
+ * @param {string} key - a unique key to identify this reducer.
+ *     The key will be used in the state object.
+ * @param {function} reducer - the reducer function
+ * @returns {function} a selector function to access the state for that reducer.
+ */
+export function registerReducer(key, reducer) {
+  return registerReducers({[key]: reducer})[key];
+}
```

Comment:
originally this is what I was using as the primary registration function, but then it ended up that the latter was the more common case. I guess I don't have a strong opinion about whether it should stick around or not. I'll remove it.